javalib/src/main/scala/java/util/regex/Matcher.scala

```diff
@@ -46,287 +45,410 @@ object Matcher {
       qrLoop(s, idx, sb).toString
     }
   }
+
 }
 
 final class Matcher private[regex] (var _pattern: Pattern,
-                                    var inputSequence: CharSequence)
+                                    var _inputSequence: CharSequence)
     extends MatchResult {
 
-  private def withRE2Regex[A](f: RE2RegExpOps => A): A =
-    _pattern.withRE2Regex(f)
+  private var currentPattern = _pattern
 
   private var hasMatch  = false
   private var hasGroups = false
   private var appendPos = 0
 
-  private var groups = withRE2Regex { re2op =>
-    val regex = re2op.ptr
-    Array.ofDim[(Int, Int)](cre2.numCapturingGroups(regex) + 1)
+  private var _regionStart = 0
+  private var _regionEnd   = _inputSequence.length
+
+  private var _groupCount = currentPattern.compiled.groupCount()
+
+  // This default differs from JVM but describes actual SN behavior.
+  // That behavior is a bug, awaiting a fix.
```

Comment:
Please file an issue and refer to it here.

Reply:
Done. Issue #1541.   

I will try to look at this after the current PR and a rest period.
This is all new work, so nothing in 0.3.8 is broken.  There are a number of regex
issues, all in obscure places, which are competing for attention.  After my brain
heals, I need to do a priority sort.